template.yaml

```diff
@@ -55,6 +55,27 @@ Conditions:
   HasDnsNames: !Not [ !Equals [ !Join [ "", !Ref DnsNames ], "" ] ]
 
 Resources:
+  CacheBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      VersioningConfiguration:
+        Status: Enabled
```

Comment:
wondering why you enabled versioning?  i don't see it used in the code?

Reply:
when you enable versioning I would expect the lambda to do something with the accumulated versions?  At the very least I would expect that you would want to give your lambda the `s3:GetObjectVersion` permission and that you use the [GetObject API](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/get_object.html) with a `VersionId`.  I'm still not sure how you plan to revert back to old versions if the lambda is not gonna do it?

Reply:
I wasn't planning on making the lambda code version-aware, it should just take the most recent version no matter what it is. and I was also planning on letting the versions accumulate to give us a full version history since the files are so small. then in the event that something goes wrong and the most recent version is corrupt or invalid, we can manually revert to a previous version by following these directions: https://docs.aws.amazon.com/AmazonS3/latest/userguide/RestoringPreviousVersions.html

Reply:
> then in the event that something goes wrong and the most recent version is corrupt or invalid, we can manually revert to a previous version by following these directions:

If that's your intention then it's fine but please make your intentions clear by adding some docs that describe the scenario and how someone can manually revert to a previous version.

Reply:
I've expanded the docs a little more, let me know whether or not it's clear enough that manually manipulating the S3 cache object can be a powerful "break-glass" admin tool when things go wrong with the upstream API.